tests/unit/lms/validation/_canvas_test.py

```diff
@@ -131,12 +145,90 @@ def test_it_raises_ValidationError_if_the_response_body_isnt_valid_json(
         }
 
     @pytest.fixture(
-        params=[canvas_list_files_response_schema, canvas_public_url_response_schema]
+        params=[
+            canvas_authenticated_users_sections_response_schema,
+            canvas_list_files_response_schema,
+            canvas_public_url_response_schema,
+        ]
     )
     def schema(self, request):
         return request.param()
 
 
+class TestCanvasAuthenticatedUsersSectionsResponseSchema:
+    def test_it_returns_a_list_of_valid_section_dicts(self, schema):
+        parsed_params = schema.parse()
+
+        assert parsed_params == [
+            {"id": 101, "name": "section_name_1"},
+            {"id": 102, "name": "section_name_2"},
+            {"id": 103, "name": "section_name_3"},
+        ]
+
+    @pytest.mark.parametrize(
+        "invalid_body,error_messages",
+        [
+            pytest.param(
+                {},
+                ["Missing data for required field."],
+                id="There must be a sections key",
+            ),
```

Comment:
It's important that we test the actual error messages from Marshmallow. Otherwise a test for one failure case could be triggering a different failure case instead and the test could be passing, when in fact the failure case that was _supposed_ to be tested might _not_ actually be failing. This kind of test mistake happens quite often if you don't use specific-enough asserts.

It so happens that the way marshmallow distinguishes one test failure from another is by its dicts-of-lists-of-strings error message format and not by, for example, raising a different exception class for each possible failure. So that's what we have to work with. (And marshmallow's design here makes good sense for a validation lib.)